### PR TITLE
Fix for wso2/product-apim#5513

### PR DESCRIPTION
**Pull request by vithu30**
Fix issue - https://github.com/wso2/product-apim/issues/5513

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -783,7 +783,9 @@ private APIDTO getAPIByAPIId(String apiId, String xWSO2Tenant) {
             }
 
             API api = apiConsumer.getAPIbyUUID(apiId, requestedTenantDomain);
-            return APIMappingUtil.fromAPItoDTO(api,requestedTenantDomain);
+            if (APIConstants.PUBLISHED.equals(api.getStatus()) || APIConstants.PROTOTYPED.equals(api.getStatus())) {
+                return APIMappingUtil.fromAPItoDTO(api, requestedTenantDomain);
+            }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java]: We should have an else statement with 403 Forbidden response.

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java]: Fixed in a1574fcf777f4bc09bad750b316b07e248eff613

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx [hunk 2]
```diff
@@ -84,21 +85,21 @@ const APIDetailsTopMenu = (props) => {
...
-                    <LaunchIcon />
-                </div>
-                <div className={classes.linkText}>View In store</div>
-            </a>
             <VerticalDivider height={70} />
+            {isVisibleInStore &&
+                <a
+                    target='_blank'
+                    rel='noopener noreferrer'
+                    href={`${window.location.origin}/store-new/apis/${api.id}/overview`}
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: Get the `store-new` context from configurations

Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: Like we have done here

https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/Publisher.jsx#L164

Comment (vithu30) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: Fixed :)

**Pull request merged**